les/api.go

```diff
@@ -352,3 +389,38 @@ func (api *PrivateLightAPI) GetCheckpointContractAddress() (string, error) {
 	}
 	return api.backend.oracle.config.Address.Hex(), nil
 }
+
+//API should be for light clients of les protocol
+type LightClientAPI struct {
+	le *LightEthereum
+}
+
+func NewLightClientAPI(le *LightEthereum) *LightClientAPI {
+	return &LightClientAPI{le}
+}
+
+func (api *LightClientAPI) GatewayFeeCache() map[string]*GatewayFeeInformation {
+	return api.le.handler.gatewayFeeCache.gatewayFeeMap
```

Comment:
As we discussed,  the gateway map is guarded by a mutex to prevent concurrent access (and the `panic` it causes). Returning the map from this method throws that protection out the window and allows the caller, in this case the RPC server code, to access to the map freely. If a write operation were to be made to the map while it is being accessed to produce the JSON serialized form, a panic would occur. If you want to return this map, it needs to be copied before being released.